config/100-deployment.yaml

```diff
@@ -120,3 +122,7 @@ spec:
                 path: oidc-token
                 expirationSeconds: 600 # Use as short-lived as possible.
                 audience: sigstore
+      - name: spire-agent-socket
+        hostPath:
```

Comment:
so based on the [k8s docs](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath) for HostPath it looks like they strongly recommend against using them for security reasons. 

>Warning:
HostPath volumes present many security risks, and it is a best practice to avoid the use of HostPaths when possible. When a HostPath volume must be used, it should be scoped to only the required file or directory, and mounted as ReadOnly.
If restricting HostPath access to specific directories through AdmissionPolicy, volumeMounts MUST be required to use readOnly mounts for the policy to be effective.

i'm hesitant to add this in to the main release -- would this still work if the directory was created in code? if not i'm wondering if maybe we should set up a second "spire-friendly" release.

Reply:
I believe Spire is looking for other options on this, and I have a lot of the same concerns. For now, I'm going to see if I can do a patch of the deployment in the test code rather than modifying this deployment here.

Reply:
sg, thanks!